### ISSUE TITLE
kdb-rm /: does not remove everything

### ISSUE THREAD

**Issue by markus2330** [ElektraInitiative/libelektra#2550]
Cascading removal of keys with globbing (_) seems to fail.

## Steps to Reproduce the Problem

```sh
kdb setmeta spec/tests/_/hello default world
kdb rm -r /tests
```

## Expected Result

That afterwards all keys are removed.

## Actual Result

The key is still there. Due to  #2456 the output is:

```
kdb ls /tests
#> /tests/_/hello
```

But the actual key is of course `spec/tests/_/hello`

## System Information

- Elektra Version: master

**Comment by kodebach** [comment]
This may be a problem in `ksCut`, because `kdb rm -r /tests` should still include everything below `spec/tests` (and therefore `spec/tests/_/hello`), no matter if globbing is used or not.

It may also be the problem that I encountered once (and never again), where somehow a cascading key ended up in the stored `dump` file and I had to manually remove it.

**Comment by markus2330** [comment]
Thank you for the confirmation and the tip where the problem is.

**Comment by markus2330** [comment]
To be solved with #3742
